org.wso2.codequalitymatrices/src/main/java/com/wso2/code/quality/metrics/GithubApiCaller.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2.code.quality.metrics;
+
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UnsupportedEncodingException;
+import java.util.Properties;
+
+import static com.wso2.code.quality.metrics.model.Constants.ACCEPT;
+import static com.wso2.code.quality.metrics.model.Constants.AUTHORIZATION;
+import static com.wso2.code.quality.metrics.model.Constants.BEARER;
+
+/**
+ * This is used for all github communications.
+ *
+ * @since 1.0.0
+ */
+public class GithubApiCaller {
```

Comment:
This looks like a util class and could be final and methods could be static as there doesn't seem to be any state to be preserved between calls.